lib/experimental/new_embed.dart

```diff
@@ -2,15 +2,237 @@
+  ConsoleTabView consoleTabView;
+
+  ExecutionService executionSvc;
+
+  NewEmbedContext context;
+
   NewEmbed() {
-    print('Created a NewEmbed.');
+    tabController = TabController();
+    for (String name in ['editor', 'test', 'console']) {
+      tabController.registerTab(
+          TabElement(querySelector('#${name}-tab'), name: name, onSelect: () {
+        editorTabView.setSelected(name == 'editor');
+        testTabView.setSelected(name == 'test');
+        consoleTabView.setSelected(name == 'console');
+      }));
+    }
+
+    testCodeButton = querySelector('#test-code');
+    runCodeButton = querySelector('#run-code');
+
+    editorTextArea = querySelector('#editor');
+    editorTabView = EditorTabView(DElement(editorTextArea));
+    consoleTabView = ConsoleTabView(DElement(querySelector('#console-view')));
+    testTabView = TestTabView(DElement(querySelector('#test-view')));
+
+    // These two will ultimately be loaded from GitHub.
+    testTabView.setTestCode(testMain);
+    editorTextArea.value = initialCode;
+
+    executionSvc = ExecutionServiceIFrame(querySelector('#frame'));
+    executionSvc.onStderr.listen((err) => consoleTabView.appendError(err));
+    executionSvc.onStdout.listen((msg) => consoleTabView.appendMessage(msg));
+    executionSvc.testResults.listen((result) {
+      consoleTabView.appendMessage(result.message);
+      if (result.success) {
+        _disableTestButton();
+      }
+    });
+
+    _initModules().then((_) => _initNewEmbed());
+  }
+
+  Future<void> _initModules() async {
+    ModuleManager modules = ModuleManager();
+
+    modules.register(DartPadModule());
+    modules.register(DartServicesModule());
+
+    await modules.start();
+  }
+
+  void _initNewEmbed() {
+    context = NewEmbedContext(
+        NewEmbedEditorFactory().createFromElement(editorTextArea));
+
+    testCodeButton.addEventListener('click', (e) => _handleRun());
+  }
+
+  // To be removed when gist-loading is integrated.
+  final testMain = '''
+void main() {
+  final str = stringify(2, 3); 
+  if (str == '2 3') {
+    _result(true, 'Test passed. Great job!');
+  } else if (str == '23') {
+    _result(false, 'Test failed. It looks like you forgot the space!');
+  } else if (str == null) {
+    _result(false, 'Test failed. Did you forget to return a value?');
+  } else {
+    _result(false, 'That\\'s not quite right. Keep trying!');
+  }
+}
+''';
+
+  // To be removed when gist-loading is integrated.
+  final initialCode = '''
+String stringify(int x, int y) {
+  // Return a formatted string here
+}
+''';
+
+  void _handleRun() {
+    final fullCode =
+        '${context.dartSource}\n$testMain\n${executionSvc.testResultDecoration}';
+    var input = CompileRequest()..source = fullCode;
+    deps[DartservicesApi]
+        .compile(input)
+        .timeout(longServiceCallTimeout)
+        .then((CompileResponse response) {
+      executionSvc.execute('', '', response.result);
+    }).catchError((e) {
+      // TODO
```

Comment:
Here and for the other TODO (modulo the ones that may be borrowed from existing code?), it's best practice to put your github user id in the TODO, in order to let future readers of the code know who has the best context for the TODO.

Reply:
Done. Thanks.